src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce<GroupsMap<T>>((acc, item) => {
+    const keyValue = String(item[key]);
```

Comment:
Consider using a more descriptive variable name instead of 'acc' for better readability. For example, you can use 'groups' or 'groupedItems'.

Reply:
Sry, cannot change variable name, error "Assignment to property of function parameter 'groups',  no-param-reassign". Also check 'gro' and 'groupedItems'
But work with 'acc', I don't know why